### DISCUSSION TITLE
Is there an API for secondary development?

### DISCUSSION THREAD

**Discussion by wangjieCode** [verdaccio/verdaccio#3194] [Q&A]
For example, realize automatic user registration, permission approval process, etc

**Comment by juanpicado** [comment]
If understood correctly, do you want to extend the API? 

Anyhow, answer is no yet, but that's the direction. I am currently working in modularize the project so could override some parts, if I manage to finish integrate Fastify also plugins would help  even more. I hope that's is what you were looking for to know more about.

**Reply by wangjieCode** [reply]
Thank you for your reply
It's like this， For example, if a new employee enters the company department, I hope to automatically create an account for him.
The reality is that I need to change the configuration file and restart the verdaccio service. Complete my needs.
Can there be a better way?

**Reply by juanpicado** [reply]
If you are using the default conf (files) nope, you need to restart. if you were using any cloud auth. plug-in then you don't. By default is like this.

**Reply by edclement** [reply]
Adding to what Juan stated, it sounds like you might want to create your own auth plugin. See: https://verdaccio.org/docs/plugin-auth#list-community-authentication-plugins for examples. That same page contains some docs near the top.

**Reply by wangjieCode** [reply]
thank you
